doc/contribute/guidelines.rst

```diff
@@ -643,53 +649,146 @@ workflow here:
    Additional information about the CI system can be found in
    `Continuous Integration`_.
 
+.. _commit-guidelines:
+
+Commit Message Guidelines
+*************************
+
+Changes are submitted as Git commits. Each commit has a *commit
+message* describing the change. Acceptable commit messages look like
+this:
+
+.. code-block:: none
+
+   [area]: [summary of change]
+
+   [Commit message body (must be non-empty)]
+
+   Signed-off-by: [Your Full Name] <[your.email@address]>
+
+You need to change text in square brackets (``[like this]``) above to
+fit your commit.
+
+Examples and more details follow.
 
-Commit Guidelines
-*****************
+Example
+=======
+
+Here is an example of a good commit message.
+
+.. code-block:: none
+
+   drivers: sensor: abcd1234: fix bus I/O error handling
+
+   The abcd1234 sensor driver is failing to check the flags field in
+   the response packet from the device which signals that an error
+   occurred. This can lead to reading invalid data from the response
+   buffer. Fix it by checking the flag and adding an error path.
 
-Changes are submitted as Git commits. Each commit message must contain:
+   Signed-off-by: Zephyr Developer <z.developer@example.com>
 
-* A short and descriptive subject line that is less than 72 characters,
-  followed by a blank line. The subject line must include a prefix that
-  identifies the subsystem being changed, followed by a colon, and a short
-  title, for example:  ``doc: update wiki references to new site``.
-  (If you're updating an existing file, you can use
-  ``git log <filename>`` to see what developers used as the prefix for
-  previous patches of this file.)
+[area]: [summary of change]
+===========================
 
-* A change description with your logic or reasoning for the changes, followed
-  by a blank line. (Every single line has to be less than 75 characters.)
+This line is called the commit's *title*. Titles must be:
 
-* A Signed-off-by line, ``Signed-off-by: <name> <email>`` typically added
-  automatically by using ``git commit -s``
+* one line
+* less than 72 characters long
+* followed by a completely blank line
 
-* If the change addresses an issue, include a line of the form::
+[area]
+  The ``[area]`` prefix usually identifies the area of code
+  being changed. It can also identify the change's wider
+  context if multiple areas are affected.
 
-      Fixes #<issue number>.
+  Here are some examples:
 
+  * ``doc: ...`` for documentation changes
+  * ``drivers: foo:`` for ``foo`` driver changes
+  * ``Bluetooth: Shell:`` for changes to the Bluetooth shell
+  * ``net: ethernet:`` for Ethernet-related networking changes
+  * ``treewide:`` for :ref:`treewide-changes`
```

Comment:
the treewide change as documented requires just the GH label no a commit prefix. If for example we have a tree-wide tree change that is in DTS or Kconfig or a macro used by each driver or subsystem such as SYS_INIT or whatever, we want to see ``dts:`` or ``kconfig:`` or ``init:``, and not ``treewide``.

Reply:
+1 on this suggestion.

Reply:
Updated accordingly.